Add support for last modified and extended status endpoints

DanWilliamsAu opened the pull request:
Adds support for a couple of endpoints that are not currently implemented in the SDK

* https://developer.smartling.com/v2.0/reference#get_projects-projectid-locales-localeid-file-status; and
* https://developer.smartling.com/v2.0/reference#get_projects-projectid-file-last-modified

dimitrystd commented on file_status_extended.go:
```diff
@@ -0,0 +1,28 @@
+}
+
+func (fs FileStatusExtended) InProgressStringCount() int {
+	return fs.NotCompletedStringCount() - fs.AwaitingAuthorizationStringCount()
```
`fs.AuthorizedStringCount` this is what translator sees. So all authorized strings are "in progress" strings

dimitrystd commented on file_status_extended.go:
```diff
@@ -0,0 +1,28 @@
+}
+
+func (fs FileStatusExtended) NotCompletedStringCount() int {
+	return fs.TotalStringCount - fs.CompletedStringCount
```
What's the difference between `InProgressStringCount` and `NotCompletedStringCount`? Both sounds like - all authorized but not translated strings yet.

dimitrystd commented on client_file_status_extended.go:
```diff
@@ -0,0 +1,29 @@
+)
+
+// GetFileStatus returns extended file status for a file for a particular locale
+func (client *Client) GetFileStatusExtended(
```
Network overhead eats significant time. I would recommend to change your application code to do a single bulk status request (all locales at once) and then work with particular locales.

If you still need this method then i would like to ask to rename it `GetFileLocaleStatus`, and `FileStatusExtended` -> `FileLocaleStatus`

DanWilliamsAu commented:
@dimitrystd thanks for your feedback.  I realise now that all the information I need is in fact on the regular status payload.  I've removed the extended endpoint code as you suggested.  I've also added a couple of helper methods to make it easier to work with.  Namely, a function to extract the FileStatusTranslation for a particular locale, and some methods for calculating awaiting authorisation numbers.

Let me know your thoughts!
